pkg/scheduler/api/unschedule_info.go

```diff
@@ -56,13 +56,24 @@ func (f *FitErrors) SetNodeError(nodeName string, err error) {
 	default:
 		fe = &FitError{
 			NodeName: nodeName,
-			Reasons:  []string{obj.Error()},
+			Status:   []*Status{{Code: Error, Reason: obj.Error()}},
 		}
 	}
 
 	f.nodes[nodeName] = fe
 }
 
+// NotHelpfulForPreemptNodes returns the set of nodes that has no help from preempting pods from it
+func (f *FitErrors) NotHelpfulForPreemptNodes() map[string]struct{} {
+	ret := make(map[string]struct{})
+	for _, node := range f.nodes {
+		if node.Status.ContainsUnschedulableAndUnresolvable() {
```

Comment:
There are some cases that prePredict or predict plugin just return err and have no statusSet returned, and here just checked status, so the nodes with only errs will also be considered when preempt, is this expected？

Reply:
Yes, we need to unify the error to `NewFitErrWithStatus` which contains status code in next pr.